### PR TITLE
Toooools

### PR DESCRIPTION
**Pull request by BarberAlec**

### CODE REVIEW

File: generated/dist-esm/models/index.d.ts [hunk 1]
```diff
@@ -8,14 +8,38 @@ export interface Conversation {
...
+    message?: string;
+    /** Any object */
+    metadata?: Record<string, unknown>;
     rating?: Rating;
+    name?: string;
+    thought?: string;
+    /** Any object */
+    input?: Record<string, unknown>;
+    observation?: string;
 }
```
Comment (BroiSatse) [inline generated/dist-esm/models/index.d.ts:23]: I am not sure if that's right. Shouldn't we have two new interfaces that extends base interface? This interface makes it completely legal to have both role and name defined on a single object

File: generated/package.json [hunk 2]
```diff
@@ -3,11 +3,11 @@
   "sdk-type": "mgmt",
   "author": "Microsoft Corporation",
   "description": "A generated SDK for ContextAPI.",
-  "version": "1.0.0-beta.1",
+  "version": "2.0.0",
```
Comment (BroiSatse) [inline generated/package.json:6]: Do we need a major version upgrade? I'd expect that 2.x.x would communicate with api v2, there's also no breaking changes

File: README.md [hunk 3]
```diff
@@ -52,12 +52,19 @@ await c.log.conversation({
...
 azure-arm: false
 add-credentials: true
 credential-scopes: all
 models-mode: msrest
 generate-metadata: true
+use:
+  - "@autorest/typescript@6.0.8"
 ```
+
+<!-- ``` yaml $(typescript)
```
Comment (agamble) [inline README.md:67]: Delete?

**Review by agamble — APPROVED** [review]

**Review by BroiSatse — APPROVED** [review]

File: generated/dist-esm/models/index.d.ts (hunk 1 above)
Comment (BarberAlec) [reply generated/dist-esm/models/index.d.ts:23]: Yup.... turns out autorest does not allow OneOf . This means that you can't mix and match in the messages array, you can only have one type. Not ideal but for the moment all messages will be of one type ...

File: generated/package.json (hunk 2 above)
Comment (BarberAlec) [reply generated/package.json:6]: Alex suggested this to make sure that the parent package in production doesn't accidentally use the new child package due to the use of ^ etc

**Pull request merged**